source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -346,7 +417,7 @@ class Filter : public Logger::Loggable<Logger::Id::ext_proc>,
   bool sent_immediate_response_ = false;
 
   // Set to true when the mergePerRouteConfig() method has been called.
-  bool route_config_merged_ = false;
+  absl::optional<FilterConfigPerRoute> route_config_merged_;
```

Comment:
I am not entirely sure that it is a recommended pattern to store per route config in the filter. cc  @yanavlasov 
That will be very helpful that if you can share more details on the issue requiring this.

(1) As we briefly mentioned [here](https://github.com/envoyproxy/envoy/pull/31431#discussion_r1432268632) before, from a quick scan  I guess you need to have `route_config_merged_`  here is because  those namespaces are now stored as pointer to vector `const std::vector<std::string>* untyped_forwarding_namespaces_{}` in ProcessState.

(2) But for per route config, it is already stored in router as a [map](https://github.com/envoyproxy/envoy/blob/7bd0c0f4fe6b30e88a77c5739a70a78bc7400ef8/source/common/router/config_impl.h#L109) and when individual filter want to use it, filter just call, for example, [resolveMostSpecificPerFilterConfig](https://github.com/envoyproxy/envoy/blob/7bd0c0f4fe6b30e88a77c5739a70a78bc7400ef8/source/common/http/utility.h#L585) to retrieve it. After all, per route config is tied with routes rather than filter.
So I am thinking if your issue is caused by something like didn't fine the route?

Reply:
When we merge the namespace configs, we sometimes construct a new vector with elements from `less-specific` and `more_specific`. This new merged vector lives on the `FilterConfigPerRoute` that is constructed in the course of `mergePerRouteConfig`, whereupon it is passed by reference into the `PorcessorState`, where a pointer to that vector is stored. After the vector goes out of scope at the return of `mergePerRouteConfig`, the memory that is pointed to by the member variable in `ProcessorState` is junk. There were two ways to solve this:
1. Store the vector on the `Filter`. This was done by storing the whole `FilterConfigPerRoute`
2. Copy the vector into `ProcessorState`. This might be preferred as it would be a smaller overall copy than copying the entire `FilterConfigPerRoute` onto the `Filter`.

I will make the switch to the second option and revert this.

Reply:
This will require copying the vectors in instead of passing by reference when constructing the `ProcessorState`. This will require copying both `FilterConfig` vectors twice for each request, plus the merged `FilterConfigPerRoute` vectors once.

Storing the newly merged `FilterConfigPerRoute` would require copying the merged vectors, plus anything else stored on the merged object. 

This latter option is probably a smaller net effect for the size of the `FilterConfigPerRoute` object currently iff this setting is configured... WDYT @tyxia

Reply:
We are creating a new `FilterConfigPerRoute` with the merged fields. This new object is created directly with the new "merge constructor" and _not_ created via `createRouteSpecificFilterConfigTyped`. The issue is that this newly created `FilterConfigPerRoute` goes out of scope at the end of `mergePerRouteConfig`'s execution, which releases the vectors created during the merge construction. Since we store a pointer to a vector in `ProcessorState`, that pointer is not good anymore when we need to access the vector. 

If the issue here is that I am using the `FilterConfigPerRoute` object in a way that is outside its intended use, I can create a new structure to perform this merge logic. But we need to store the vectors on the `Filter` or `ProcessorState` since they are constructed during the merge, which takes place in the request path.